### PR TITLE
WIP: sneak dockerization + config module + node 7.5

### PR DESCRIPTION
**Pull request by sneak**

### CODE REVIEW

**Comment by sneak** [comment]
Don't merge this yet, I still need to do more testing.

File: db/models/index.js [hunk 1]
```diff
@@ -3,13 +3,15 @@ var path = require('path');
...
 var env = process.env.NODE_ENV || 'development';
-var config = require(__dirname + '/../../config').default.db;
+var config = require('config');
 var db = {};
 
-if (config.use_env_variable) {
-    var sequelize = new Sequelize(process.env[config.use_env_variable]);
+var dbConfig = config.get('db');
+
+if (dbConfig.use_env_variable) {
```
Comment (sneak) [inline db/models/index.js]: make using an env variable for the DATABASE_URL default

File: config/client_config.js [hunk 2]
```diff
@@ -1,43 +1,51 @@
...
-export const APP_NAME_UPPERCASE = 'STEEMIT'
-export const APP_ICON = 'steem'
-export const APP_DOMAIN = 'steemit.com'
-export const LIQUID_TOKEN = 'Steem'
+export const APP_NAME_LATIN = "Steemit";
+export const APP_NAME_UPPERCASE = "STEEMIT";
+export const APP_ICON = "steem";
+// FIXME figure out best way to do this on both client and server from env
+// vars. client should read $STM_Config, server should read config package.
+export const APP_DOMAIN = "steemit.com";
```
Comment (sneak) [inline config/client_config.js]: talk to @valzav and @roadscape about how to best parameterize this

File: server/server.js [hunk 3]
```diff
@@ -136,50 +169,103 @@ useRedirects(app);
...
-        host: 'http://0.0.0.0:' + PORT,
-        map: (filePath) => 'assets/' + filePath
+if (env === "development") {
+    const webpack_dev_port = process.env.PORT
+        ? parseInt(process.env.PORT) + 1
+        : 8081;
+    const proxyhost = "http://0.0.0.0:" + webpack_dev_port;
+    console.log("proxying to webpack dev server at " + proxyhost);
+    const proxy = require("koa-proxy")({
+        host: "http://0.0.0.0:" + webpack_dev_port,
```
Comment (sneak) [inline server/server.js]: use proxyhost here

**Comment by sneak** [comment]
AFAICT this is ready - could you all please test the hell out of it?

**Comment by sneak** [comment]
particularly important is testing the DATABASE_URL env var with mysql, i think. (we lost `db/console.js` until i add parsing of the DATABASE_URL back in...)

**Comment by roadscape** [comment]
Ideally formatting changes are confined to their own commit (or just separate PR) for big changes like this

File: server/server.js [hunk 4]
```diff
@@ -51,6 +51,13 @@ session(app, {
 app.use(mount(grant));
 app.use(flash({ key: 'flash' }));
 
+function convertEntriesToArrays(obj) {
```
Comment (sneak) [inline server/server.js]: this is way better than the way i did it! 👍

File: package.json [hunk 5]
```diff
@@ -40,7 +40,7 @@
     "bytes": "^2.4.0",
     "classnames": "^2.2.5",
     "co-body": "^4.2.0",
-    "config": "^1.25.1",
+    "@steem/config": "^1.25.1",
```
Comment (sneak) [inline package.json]: @valzav i think this should remain the same - it's causing 'yarn' to fail to install/build.

**Comment by sneak** [comment]
utACK - looks good! @jredbeard make sure it connects to mysql in staging using the sql connection url syntax.

**Comment by sneak** [comment]
@valzav please test and merge.

**Pull request merged**